File: src/main/java/jndev/bonebot/Responder.java
package jndev.bonebot;

import net.dv8tion.jda.api.entities.Activity;
import net.dv8tion.jda.api.entities.Message;

import java.io.File;
import java.io.FileNotFoundException;
import java.util.ArrayList;
import java.util.Scanner;

/**
 * responder to words and phrases in a message
 */
public class Responder {
    
    /**
     * list of phrases loaded from the responses file
     */
    private static final ArrayList<String> responses = new ArrayList<>();
    
    /**
     * load all data from responses file
     */
    public static void loadData() {
        try {
            Scanner fileScanner = new Scanner(new File("responses.txt"));
            responses.clear();
            while (fileScanner.hasNextLine()) responses.add(fileScanner.nextLine());
            fileScanner.close();
            responses.trimToSize();
        } catch (FileNotFoundException e) {
            Logger.log(e);
        }
    }
    
    /**
     * respond to a message if a trigger phrase is said
     *
     * @param message message to check and respond to
     */
    public static void respond(Message message) {
        String msg = message.getContentRaw().toLowerCase();
        for (String phrase : responses) {
            String[] triggerAndPhrase = phrase.split(" // ");
            String[] triggers = triggerAndPhrase[0].split(" / ");
            int count = 0;
            for (String trigger : triggers) {
                if (msg.contains(trigger.toLowerCase())) count++;
            }
            if (count == triggers.length) {
                message.getJDA().getPresence().setActivity(Activity.playing("Discord"));
                message.getChannel().sendMessage(triggerAndPhrase[1]
                        .replace("$USER$", message.getAuthor().getAsMention())).queue();
            }
        }
    }
    
}

File: src/main/kotlin/jndev/bonebot/Responder.kt
package jndev.bonebot

import net.dv8tion.jda.api.entities.Message
import java.util.*

/**
 * responder to words and phrases in a message
 *
 * @author JNDev (Jeremaster101)
 */
object Responder {
    /**
     * list of phrases loaded from the responses file
     */
    val responses = ArrayList<String>()

    /**
     * last time the responder sent a message in milliseconds
     */
    private var prevTime = 0L

    /**
     * respond to a message if a trigger phrase is said
     *
     * @param message message to check and respond to
     */
    fun respond(message: Message) {
        val msg = message.contentRaw.toLowerCase()
        for (phrase in responses) {
            val triggerAndPhrases = phrase.split(" // ").toTypedArray()
            val triggers = triggerAndPhrases[0].split(" / ").toTypedArray()
            var count = 0
            for (trigger in triggers) {
                if (msg.contains(trigger.toLowerCase())) count++
            }
            if (count == triggers.size && (System.currentTimeMillis() - prevTime) >= Config.responseCooldown * 1000) {
                prevTime = System.currentTimeMillis()
                for (i in 1 until triggerAndPhrases.size) message.channel.sendMessage(
                    triggerAndPhrases[1]
                        .replace("\$USER$", message.author.asMention)
                ).queue()
            }
        }
    }
}